spec/controllers/appointments_controller_spec.rb

```diff
@@ -0,0 +1,141 @@
+    }
+  end
+
+  describe 'GET /appointments' do
+    it 'successfully renders the index view for the user\'s appointments' do
+      user = User.create!(user_attributes)
+      login_user(user)
+
+      get :index, params: { user_id: user.id }
+      expect(response).to be_successful
+    end
+  end
+
+  describe 'GET #show' do
+    it 'successfully renders the show view for the specified appointment' do
+      user = User.create!(user_attributes)
+      login_user(user)
+
+      availability = Availability.create!(availability_attributes.merge(id: 1))
+      appointment = user.appointments.create!(appointment_attributes)
+
+      get :show, params: { user_id: user.id, id: appointment.id }
+
+      expect(response).to be_successful
+    end
+  end
+
+  describe 'GET #new' do
+    it 'successfully renders the new appointment form and initializes a new appointment' do
+      user = User.create!(user_attributes)
+      login_user(user)
+
+      # Create availability slots for the specific date
+      Availability.create!(availability_attributes)
+      
+      get :new, params: { user_id: user.id, day: date }
+      expect(response).to be_successful
+      
+      # Ensure appointment is initialized
+      expect(assigns(:appointment)).to be_a_new(Appointment)
+    end
+  end
+
+  describe 'GET /edit' do
+    it 'successfully renders the edit form for the specified appointment' do
+      user = User.create!(user_attributes)
+      login_user(user)
+  
+      availability = Availability.create!(availability_attributes.merge(id: 1))
+      appointment = user.appointments.create!(appointment_attributes)
+
+      get :edit, params: { user_id: user.id, id: appointment.id}
+      expect(response).to be_successful
+    end
+  end
+
+  describe 'POST #create' do
+    it 'creates a new appointment and redirects to the appointment show page' do
+      user = User.create!(user_attributes)
+      login_user(user)
+
+      # Create availability slots for the specific date
+      Availability.create!(availability_attributes.merge(id: 1))
+      
+      post :create, params: { user_id: user.id, appointment: appointment_attributes }
+      
+      expect(response).to redirect_to(user_appointment_path(user, Appointment.last))
+    end
+  end
+
+  describe 'PATCH #update' do
+    it 'updates the specified appointment and redirects to the appointment show page' do
+      user = User.create!(user_attributes)
+      login_user(user)
+
+      availability = Availability.create!(availability_attributes.merge(id: 1))
+      appointment = user.appointments.create!(appointment_attributes)
+      
+      patch :update, params: { user_id: user.id, id: appointment.id, appointment: new_attributes }
+      
+      expect(response).to redirect_to(user_appointment_path(user, appointment))
+    end
+  end
+
+  describe 'DELETE #destroy' do
+    it 'destroys the specified appointment and redirects to the appointments index' do
+      user = User.create!(user_attributes)
```

Comment:
I repeat the code from lines 126 to 130 a few times in this spec file. Is there a way to make this better?

Reply:
Below line 7 in the beginning of your spec you can have a 
```
before(:all)
  user = User.create!(user_attributes)
  login_user(user)
do
```
This will load it before all of your specs run.

then in each describe you can add a before do for 
```
before do
   availability = Availability.create!(availability_attributes.merge(id: 1))
    appointment = user.appointments.create!(appointment_attributes)
end
```
since this does not appear in all of your describe